### DISCUSSION TITLE
Dynamic queryStrings with queryString()

### DISCUSSION THREAD

**Discussion by m4tr1ck** [livewire/livewire#5606] [1. Help] [answered]
I try to get dynamic queryStrings with the functionality of alias. 
Like following: 
```
public string $suffix = '';

public function queryString()
    {
        return [
            'page' => ['except' => 1, 'as' => $this->suffix.'page'],
            'some_id' => ['except' => 0, 'as' => $this->suffix.'some_id'],
        ];
    }
```
After the initialization of the component changing one of both properties works fine. Trying to reload the page or access it with the defined query strings in the uri is not working like intended. If the variable suffix got a default value it will be taken on the first boot up of the component in the first cycle for the hydration after that the set value for suffix will be loaded. 
I expected that it would load the set value for the var suffix and after that the function queryString() would be used to hydrate the rest of it, but actually it will load with the default value and after loading the set value of var suffix it won't update the vars inside the class with the correct queryString values. 

The component is getting initialized without and once with a set value for suffix. I tried make use of the mount function but nothing changed. 

Expected behavior: uri?some_id=1&delsome_id=3 => comp. with default suffix got 1 and comp. with 'del' suffix got 3. 
Actual behavior: both comp. load the value 1. 

Updating values of the queryString with this dynamic alias is working like intended. 

I also checked the value of suffix for a comp with a given value for it and it got resolved 2 times with the default value for the variable and after that the hydration gets the correct value of it.

Someone got some Ideas why this is happening or some other advises? 
I only try to remove the need of creating 2 mostly similar classes with the difference in 5 lines of code.

**Comment by joshhanley** [comment, marked as answer]
@m4tr1ck unless I'm misunderstanding, then that is how I'd expect it to work, as Livewire has no way to know that you've defined a new `suffix`. Livewire loads data from the queryString before mount, so the passed in data to the component wouldn't be available yet, hence why suffix is an empty string.

Maybe there is another way around it though. Can you explain in detail what you are trying to achieve from doing this and why? Then we can see if there is another option.

Hope this helps!

**Reply by m4tr1ck** [reply]
Thank you for the explanation, that would explain the whole behavior.

What I wanted to archive was that I can use a single class to manage a collection of a model with pagination, but that file will allow to use a suffix or some other variable to define that the softdeleted models get loaded instead. The only changes would be about 2 html tags and inside the class about 4 lines of code. 

The other Way around would be to make use of the class as a parent class and extend it for the softdeleted variant with the changed queryString.

**Reply by m4tr1ck** [reply]
With extending the original class and changing the queryString worked for me.
